source/connect/csot.txt

```diff
@@ -1,210 +1,241 @@
+10 seconds:
 
 .. code-block:: python
 
    with pymongo.timeout(10):
-       coll.insert_one({"name": "Nunu"})
-       coll.find_one({"name": "Nunu"})
+       coll.insert_one({"name": "Yngwie"})
+       coll.find_one({"name": "Yngwie"})
 
-If you nest calls to the ``~pymongo.timeout()`` method, the deadline of the inner method
-call is capped by the deadline of the outer method call. The deadline can only be
-shortened, not extended.
+Nested Timeout Blocks
+~~~~~~~~~~~~~~~~~~~~~
 
-When execution enters a nested ``timeout`` block, the outer ``timeout`` deadline is paused.
-When execution exits the nested ``timeout`` block, the previous ``timeout`` deadline resumes.
-The following example shows how nested calls to the ``timeout()`` method interact:
-
-.. code-block:: python
+When execution enters a nested timeout block, the outer timeout is
+paused. When execution exits the nested timeout block, the previous timeout
+resumes.
 
-   with pymongo.timeout(5):
-       coll.find_one()  # Uses the 5 second deadline.
-       with pymongo.timeout(3):
-           coll.find_one() # Uses the 3 second deadline.
-       coll.find_one()  # Uses the original 5 second deadline.
-       with pymongo.timeout(10):
-           coll.find_one()  # Still uses the original 5 second deadline.
-       coll.find_one()  # Uses the original 5 second deadline.
+.. important::
 
-Timeout Errors
---------------
+   A nested timeout block can shorten the deadline of an outer timeout block,
+   but it can't extend it.
 
-When execution enters a ``~pymongo.timeout()`` ``with`` block, a deadline is set
-for the entire block. When that deadline is exceeded, any blocking {+driver-short+}
-operation will raise a timeout exception. For example:
+The following example shows how to use nested calls to the ``timeout()`` method:
 
 .. code-block:: python
+   :linenos:
 
-   try:
+   with pymongo.timeout(5):
+       collection.find_one()
+       with pymongo.timeout(3):
+           collection.find_one()
+       collection.find_one()
        with pymongo.timeout(10):
-           coll.insert_one({"name": "Nunu"})
-           time.sleep(10)
-           # The deadline has now expired, the next operation will raise
-           # a timeout exception.
-           coll.find_one({"name": "Nunu"})
-   except PyMongoError as exc:
-       if exc.timeout:
-           print(f"block timed out: {exc!r}")
-       else:
-           print(f"failed with non-timeout error: {exc!r}")
-
-The ``pymongo.errors.PyMongoError.timeout`` property (added in PyMongo 4.2)
-is ``True`` when the error was caused by a timeout and ``False`` otherwise.
+           collection.find_one()
+       collection.find_one()
+
+In the preceding example, the code performs the following steps:
+
+- **Line 1**: Creates a five-second timeout block.
+- **Line 2**: Calls the ``find_one()`` method inside the five-second timeout block. The server must
+   complete this operation within the five-second timeout.
```

Comment:
There's a rendering issue here causing the hole line to bold. I'm not sure if this will fix it, but maybe?

```suggestion
- **Line 2**: Calls the ``find_one()`` method inside the five-second timeout block. The server must
  complete this operation within the five-second timeout.
```